### PR TITLE
add tc

### PR DESCRIPTION
**Pull request by HookedBehemoth**

### CODE REVIEW

File: nx/include/switch/services/tc.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+/// Exit tc.
+void tcExit(void);
+
+/// Getc the Service for tc.
+Service* tcGetServiceSession(void);
+
+Result tcEnableFanControl(void);
+Result tcDisableFanControl(void);
+Result tcIsFanControlEnabled(bool *status);
+/// Only available on [5.0.0+]
```
Comment (yellows8) [inline nx/include/switch/services/tc.h]: period

File: nx/source/services/tc.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Result tcEnableFanControl(void) {
+    return _tcNoInNoOut(6);
+}
+
+Result tcDisableFanControl(void) {
+    return _tcNoInNoOut(7);
+}
+
+Result tcIsFanControlEnabled(bool *status) {
+    struct {
```
Comment (yellows8) [inline nx/source/services/tc.c]: Replace with just `u8 out;`

File: nx/source/services/tc.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        u8 status;
+    } out;
+
+    Result rc = serviceDispatchOut(&g_tcSrv, 8, out);
+    if (R_SUCCEEDED(rc)) *status = out.status;
```
Comment (yellows8) [inline nx/source/services/tc.c]: `if (R_SUCCEEDED(rc) && status) *status = out & 1;`

File: nx/source/services/tc.c [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+Result tcGetSkinTemperatureMilliC(s32 *skinTemp) {
+    if (hosversionBefore(5,0,0))
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    struct {
+        s32 skinTemp;
+    } out;
+
+    Result rc = serviceDispatchOut(&g_tcSrv, 9, out);
```
Comment (yellows8) [inline nx/source/services/tc.c]: Just use *skinTemp directly.

File: nx/include/switch/services/tc.h [hunk 5]
```diff
@@ -0,0 +1,25 @@
+/**
+ * @file tc.h
+ * @brief Temperature control (tc) service IPC wrapper.
+ * @author yellows8
```
Comment (yellows8) [inline nx/include/switch/services/tc.h]: Adjust that.

File: nx/include/switch/services/tc.h [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+#include "../types.h"
+#include "../sf/service.h"
+
+/// Initialize tc.
+Result tcInitialize(void);
+
+/// Exit tc.
+void tcExit(void);
+
+/// Getc the Service for tc.
```
Comment (yellows8) [inline nx/include/switch/services/tc.h]: Gets

Comment (HookedBehemoth) [reply nx/include/switch/services/tc.h]: sed -i 's/ts/tc/g'
^^

**Pull request merged**